core/iwasm/common/wasm_memory.c

```diff
@@ -737,8 +656,15 @@ wasm_runtime_validate_app_addr(WASMModuleInstanceCommon *module_inst_comm,
     /* boundary overflow check */
     if (size > max_linear_memory_size
         || app_offset > max_linear_memory_size - size) {
-        goto shared_heap_bound_check;
+        goto fail;
+    }
+
+#if WASM_ENABLE_SHARED_HEAP != 0
+    if (is_app_addr_in_shared_heap(module_inst_comm, memory_inst->is_memory64,
```

Comment:
If the app_offset is in shared heap, it's seems that the check will go to fail at line 659, and is_app_addr_in_shared_heap will not work. Move this before line 657?

Reply:
yes, thanks, done.